openstack-device.tf

```diff
@@ -157,6 +157,22 @@ resource "openstack_compute_secgroup_v2" "AUFN" {
   }
 }
 
+data "openstack_dns_zone_v2" "lab_zone" {
+  count = var.dns_zone_name != null ? 1 : 0
+  name  = var.dns_zone_name
+}
+
+resource "openstack_dns_recordset_v2" "lab_dns" {
+  count       = var.dns_zone_name != null ? var.lab_count : 0
+  zone_id     = data.openstack_dns_zone_v2.lab_zone[0].id
+  name        = format("%s-lab-%02d.%s", var.lab_prefix, count.index, var.dns_zone_name)
+  type        = "A"
+  ttl         = 300
+  records     = [openstack_compute_instance_v2.lab[count.index].network[0].fixed_ip_v4]
+
+  depends_on = [openstack_compute_instance_v2.lab]
```

Comment:
Should this depend only on the relevant instance? Otherwise would a change in any instance cause all records to get recreated?

Reply:
Not sure if there is much benefit it should only change things that need changing.

However now looking back at this I think we could drop `depends_on` as `records` should create dependency between `lab_dns` and `lab`.

Reply:
+1